src/anyio/_backends/_asyncio.py

```diff
@@ -2034,9 +2034,14 @@ def run_async_from_thread(
         args: tuple[Any, ...],
         token: object,
     ) -> T_Retval:
+        @wraps(func)
+        async def wrapper() -> T_Retval:
+            sniffio.current_async_library_cvar.set("asyncio")
+            return await func(*args)
+
         loop = cast(AbstractEventLoop, token)
         f: concurrent.futures.Future[T_Retval] = asyncio.run_coroutine_threadsafe(
-            func(*args), loop
+            wrapper(), loop
         )
         return f.result()
```

Comment:
This is a rather unpleasant solution. Have you explored options that didn't involve a wrapper coroutine?

Reply:
This also passes the tests:
```python
    @classmethod
    def run_async_from_thread(
        cls,
        func: Callable[..., Awaitable[T_Retval]],
        args: tuple[Any, ...],
        token: object,
    ) -> T_Retval:
        loop = cast(AbstractEventLoop, token)
        ctx = copy_context()
        ctx.run(sniffio.current_async_library_cvar.set, "asyncio")
        f: concurrent.futures.Future[T_Retval] = ctx.run(
            asyncio.run_coroutine_threadsafe, 
            func(*args),
            loop
        )
        return f.result()
```